Framework/Algorithms/test/ReflectometryBackgroundSubtractionTest.py

```diff
@@ -0,0 +1,140 @@
+                'BackgroundCalculationMethod' : 'Polynomial',
+                'DegreeOfPolynomial' : '0',
+                'OutputWorkspace': 'output'}
+        output = self._assert_run_algorithm_succeeds(args)
+        for i in range(0, output.getNumberHistograms()):
+            if i == 3 or i == 4:
+                for itr in range(0, output.blocksize()):
+                    self.assertEquals(3.0, output.dataY(i)[itr])
+            else:
+                for itr in range(0, output.blocksize()):
+                    self.assertEquals(0.0, output.dataY(i)[itr])
+
+    def test_output_AveragePixelFit(self):
+        """
+            Check that the algorithm the correct output using background method AveragePixelFit
+        """
+        args = {'InputWorkspace' : 'workspace_with_peak', 
+                'InputWorkspaceIndexSet' : '0-7',
+                'BackgroundCalculationMethod' : 'AveragePixelFit',
+                'PeakRange' : '3,4',
+                'SumPeak' : False,
+                'OutputWorkspace': 'output'}
+        output = self._assert_run_algorithm_succeeds(args)
+        for i in range(0, output.getNumberHistograms()):
+            if i == 3 or i == 4:
+                for itr in range(0, output.blocksize()):
+                    self.assertEquals(3.0, output.dataY(i)[itr])
+            else:
+                for itr in range(0, output.blocksize()):
+                    self.assertEquals(0.0, output.dataY(i)[itr])
+
+    def test_Polynomial_error_for_single_spectra(self):
+        args = {'InputWorkspace' : 'workspace_with_peak',
+                'InputWorkspaceIndexSet' : '3',
+                'BackgroundCalculationMethod' : 'Polynomial'}
+        self._assert_run_algorithm_throws(args)
+
+    def test_AveragePixelFit_error_for_single_spectra(self):
+        args = {'InputWorkspace' : 'workspace_with_peak',
+                'InputWorkspaceIndexSet' : '3',
+                'BackgroundCalculationMethod' : 'AveragePixelFit',
+                'PeakRange' : '3,4',
+                'OutputWorkspace': 'output'}
+        self._assert_run_algorithm_throws(args)
+
+    def test_AveragePixelFit_error_peakRange_outside_spectra(self):
+        args = {'InputWorkspace' : 'workspace_with_peak',
+                'InputWorkspaceIndexSet' : '1-7',
+                'BackgroundCalculationMethod' : 'AveragePixelFit',
+                'PeakRange' : '3,9',
+                'OutputWorkspace': 'output'}
+        self._assert_run_algorithm_throws(args)
+
+    def _assert_run_algorithm_succeeds(self, args):
+        """ Run the algorithm with the given args and check it succeeds """
+        alg = create_algorithm('ReflectometryBackgroundSubtraction', **args)
+        assertRaisesNothing(self, alg.execute)
+        self.assertTrue(mtd.doesExist('output'))
+        return mtd['output']
+
+    def _assert_run_algorithm_throws(self, args = {}):
+        """Run the algorithm with the given args and check it throws"""
+        throws = False
```

Comment:
One last leftover, otherwise looks good.

Reply:
Ah yes, didn't see that one